stdlib/public/core/Collection.swift

```diff
@@ -123,10 +123,9 @@ extension IndexingIterator: IteratorProtocol, Sequence {
   @inlinable
   @inline(__always)
   public mutating func next() -> Elements.Element? {
-    if _position == _elements.endIndex { return nil }
-    let element = _elements[_position]
-    _elements.formIndex(after: &_position)
-    return element
+    guard _position < _elements.endIndex else { return nil }
+    defer { _elements.formIndex(after: &_position) }    
```

Comment:
I'd suggest changing one thing at a time when experimenting like this. Switching this to a `defer` could lead to significantly different codegen depending on what the optimizer does with it, muddying measurement of the `<` change.